### PR TITLE
Refactor and clean up Scholly service

### PR DESCRIPTION
**Pull request by aksh0001**
# Description 
Scholly service's building workflow needed a revamp to ensure logs were being outputted correctly. The overall code has been improved for readability as well as conciseness. Performance is slightly better (because we're doing away with the call to `which` to find the absolute path to `npm`), but further optimization-related work will be in future PRs. I've also updated our lockfiles as previous commits made changes that were using an old lockfile and running `npm` produced warnings about this.

# Type of change 
- [x] Bug fix
- [ ] New feature
- [x] Refactoring
- [ ] Documentation 

# Problem

See desc.
 

# Solution description

- Set minimum logging level to `debug`
- Refactor subprocess creation to use `execFile` API (higher-level wrapper around `spawn`)

# Tests 
All unit tests pass? Y

Builds successfully? Y

# Risk
Low

### CODE REVIEW

File: scholly/service/core/build.js [hunk 1]
```diff
@@ -28,43 +30,42 @@ const BUILD_TIMEOUT = 10000;
...
-        [REACT_APP_TEAM_HOMEPAGE]: JSON.stringify(data.teamHomepage),
-        [REACT_APP_WEB_PAGES]: JSON.stringify(data.webPages),
-        [REACT_APP_TEAM_ACHIEVEMENTS]: JSON.stringify(data.teamAchievements),
+  let stdout; // Capture stdout of the spawned subprocess.
+  let stderr; // Capture stderr of the spawned subprocess.
+  try {
+    ({
+      stdout,
+      stderr,
+    } = await exec(
```
Comment (aksh0001) [inline scholly/service/core/build.js:39]: Relevant docs:

- https://nodejs.org/api/child_process.html#child_process_child_process
- https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original

**Review by jfan0010 — APPROVED** [review]

**Review by rpra0006 — APPROVED** [review]

**Comment by kdav108** [comment]
Are we making any changes to scholly base package dependencies?

**Comment by kdav108** [comment]
Also, I'm getting the following error when deploying:
Screen Shot 2021-09-23 at 6 52 31 pm

Am I missing something?

**Comment by jfan0010** [comment]
> Also, I'm getting the following error when deploying:
> Screen Shot 2021-09-23 at 6 52 31 pm
> 
> Am I missing something?

Got the same error now as well. That's weird cause I tried to deploy this morning and everything was fine.

**Comment by pmyip3** [comment]
> Also, I'm getting the following error when deploying:
> Screen Shot 2021-09-23 at 6 52 31 pm
> 
> Am I missing something?

I remember you are using mac Kunj? I am having this error too thought it only happens to the window

**Comment by vgoh1** [comment]
I'm joining the error train:
image

**Comment by aksh0001** [comment]
> Are we making any changes to scholly base package dependencies?

The changes are lockfile upgrades as I believe we were using an old lockfile (possibly due to an unintended Windows PR change) that threw warnings inside the npm console
see https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json#lockfileversion

**Comment by aksh0001** [comment]
Fixed those issues, could you retry? Windows support should be fixed, but let me know if it still doesn't work.

**Review by kdav108 — APPROVED** [review]
It's working for me now! Would be good to get a windows user to try as well though.

**Comment by jfan0010** [comment]
The deployment is working on my Windows PC as well.

**Pull request merged**